app/controllers/application_controller.rb

```diff
@@ -1,3 +1,13 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
+
+  def require_user
+    redirct_to login_path unless current_user
+  end
+
+  def current_user
+    User.find(session[:user_id]) if session[:user_id]
```

Comment:
Good practice is to memoize the user object that is found by `User.find`. You are going to call that `current_user` method per request a lot of times.